### PR TITLE
New Resource `azurerm_network_manager_security_admin_configuration`, `azurerm_network_manager_admin_rule_collection`, `azurerm_network_manager_admin_rule`

### PR DESCRIPTION
**Pull request by teowa**
These three resources have strong dependency: configuration -> collection -> rule. from doc:
> A security admin configuration contains a set of rule collections. Each rule collection contains one or more security admin rules.
 
And they don't have many properties (about 20 totally), so maybe we can review these together?

Reference:
- [Security admin rules in Azure Virtual Network Manager](https://learn.microsoft.com/en-us/azure/virtual-network-manager/concept-security-admins)
- [Swagger File](https://github.com/Azure/azure-rest-api-specs/blob/4c8d1b96e36c12bf3c9c260e5de21df8d417bf2b/specification/network/resource-manager/Microsoft.Network/stable/2022-07-01/networkManagerSecurityAdminConfiguration.json)

Test:
```hcl
TF_ACC=1 go test -v ./internal/services/network -parallel 20 -test.run=TestAccNetworkManager -timeout 1440m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccNetworkManager
=== RUN   TestAccNetworkManager/SecurityAdminConfiguration
=== RUN   TestAccNetworkManager/SecurityAdminConfiguration/basic
=== RUN   TestAccNetworkManager/SecurityAdminConfiguration/complete
=== RUN   TestAccNetworkManager/SecurityAdminConfiguration/update
=== RUN   TestAccNetworkManager/SecurityAdminConfiguration/requiresImport
=== RUN   TestAccNetworkManager/AdminRuleCollection
=== RUN   TestAccNetworkManager/AdminRuleCollection/complete
=== RUN   TestAccNetworkManager/AdminRuleCollection/update
=== RUN   TestAccNetworkManager/AdminRuleCollection/requiresImport
=== RUN   TestAccNetworkManager/AdminRuleCollection/basic
=== RUN   TestAccNetworkManager/AdminRule
=== RUN   TestAccNetworkManager/AdminRule/basic
=== RUN   TestAccNetworkManager/AdminRule/complete
=== RUN   TestAccNetworkManager/AdminRule/update
=== RUN   TestAccNetworkManager/AdminRule/requiresImport
--- PASS: TestAccNetworkManager (2438.84s)
    --- PASS: TestAccNetworkManager/SecurityAdminConfiguration (704.29s)
        --- PASS: TestAccNetworkManager/SecurityAdminConfiguration/basic (171.12s)
        --- PASS: TestAccNetworkManager/SecurityAdminConfiguration/complete (179.91s)
        --- PASS: TestAccNetworkManager/SecurityAdminConfiguration/update (189.99s)
        --- PASS: TestAccNetworkManager/SecurityAdminConfiguration/requiresImport (163.27s)
    --- PASS: TestAccNetworkManager/AdminRuleCollection (790.21s)
        --- PASS: TestAccNetworkManager/AdminRuleCollection/complete (253.85s)
        --- PASS: TestAccNetworkManager/AdminRuleCollection/update (217.73s)
        --- PASS: TestAccNetworkManager/AdminRuleCollection/requiresImport (123.94s)
        --- PASS: TestAccNetworkManager/AdminRuleCollection/basic (194.69s)
    --- PASS: TestAccNetworkManager/AdminRule (944.34s)
        --- PASS: TestAccNetworkManager/AdminRule/basic (246.42s)
        --- PASS: TestAccNetworkManager/AdminRule/complete (199.50s)
        --- PASS: TestAccNetworkManager/AdminRule/update (304.67s)
        --- PASS: TestAccNetworkManager/AdminRule/requiresImport (193.75s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/network       2440.082s
```

### CODE REVIEW

File: website/docs/r/network_manager_admin_rule.html.markdown [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `name` - (Required) Specifies the name which should be used for this Network Manager Admin Rule. Changing this forces a new Network Manager Admin Rule to be created.
+
+* `admin_rule_collection_id` - (Required) Specifies the ID of the Network Manager Admin Rule Collection. Changing this forces a new Network Manager Admin Rule to be created.
+
+* `access` - (Required) Specifies the access allowed for this Network Manager Admin Rule. Possible values are `Allow`, `AlwaysAllow`, and `Deny`.
```
Comment (katbyte) [inline website/docs/r/network_manager_admin_rule.html.markdown:80]: should this be action not access to match other resources?

File: website/docs/r/network_manager_admin_rule.html.markdown [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+* `direction` - (Required) Indicates if the traffic matched against the rule in inbound or outbound. Possible values are `Inbound` and `Outbound`.
+
+* `priority` - (Required) The priority of the rule. Possible values are integer between `1` and `4096`. The priority number must be unique for each rule in the collection. The lower the priority number, the higher the priority of the rule.
+
+* `protocol` - (Required) Specifies which network protocol this Network Manager Admin Rule applies to. Possible values are `Ah`, `Any`, `Esp`, `Icmp`, `Tcp`, and `Udp`.
+
+* `description` - (Optional) A description of the Network Manager Admin Rule.
+
+* `destination_port_ranges` - (Optional) A list of string specifies the destination port ranges.
```
Comment (katbyte) [inline website/docs/r/network_manager_admin_rule.html.markdown:90]: could we include some examples?

File: internal/services/network/network_manager_admin_rule_resource.go [hunk 3]
```diff
@@ -0,0 +1,438 @@
...
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"destination_port_ranges": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Schema{
+				Type: pluginsdk.TypeString,
+			},
```
Comment (katbyte) [inline internal/services/network/network_manager_admin_rule_resource.go:117]: can we validate this?

File: internal/services/network/network_manager_admin_rule_resource.go [hunk 4]
```diff
@@ -0,0 +1,438 @@
...
+				},
+			},
+		},
+
+		"source_port_ranges": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Schema{
+				Type: pluginsdk.TypeString,
+			},
```
Comment (katbyte) [inline internal/services/network/network_manager_admin_rule_resource.go:148]: can we validate this?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: website/docs/r/network_manager_admin_rule.html.markdown (hunk 1 above)
Comment (teowa) [reply website/docs/r/network_manager_admin_rule.html.markdown:80]: changed to `action`

**Review by katbyte — APPROVED** [review]
Thanks @teowa LGTM ⭐

**Pull request merged**